### ISSUE TITLE
Question regarding agents and sessions APIs

### ISSUE THREAD

**Issue by Minutis** [meta-llama/llama-stack#197]
Hello,

I have a question regarding the API with Agents. I did not find a way to get an agent that is already created. Or list the agents available. Did I miss something or is the agent not supposed to be a longer lived entity (than let's say a method call)? I imagine, that you configure agents and then use them throughout your application.

Same question regarding sessions. Although I think session lifespan is shorter than agent it still would make sense to be able to get existing sessions for w/e the purpose it could be. I see that it's possible to fetch single session but I don't see a way to list all sessions.

**Comment by yanxi0830** [comment]
Yes, the ability to get session and list all sessions associated with an agent is not implemented, and is a functionality we should implement. 

We currently identify agents via `agent_id`, and sessions for the agent via `session_id`. Example snippet: https://github.com/meta-llama/llama-stack-apps/blob/36b04bf3cfb79cbc4383cf6609332da34a6b48c0/common/client_utils.py#L179-L195

**Comment by tamdogood** [comment]
i am working on this. will put up a PR soon

**Comment by wukaixingxp** [comment]
@Minutis [this notebook](https://llama-stack.readthedocs.io/en/latest/building_applications/agent.html#core-concepts) may be helpful to explain agents and sessions concepts
